articles/ros_node_idl.md

```diff
@@ -113,43 +118,59 @@ Another is the fact that, as soon as the node is running, RCL itself (or another
 How do upstream packages specify their interface requirements?
 Through a high-level description of all the actions, parameters, services and topics, provided or required, by each node within the package.
 
-The package interface is defined in a separate [XML][xml_wiki] file and exported from the `package.xml` using [REP 149's export mechanism][rep149_export], thereby avoiding pollution of the package manifest.
+The package interface is defined in a separate [XML][xml_wiki] file with suffix `.nodl.xml`. This XML file is exported to the [ament index][ament_index], either manually in the case of python projects or with a helper CMake macro.
```

Comment:
Haha, I know this is awful, but: every time you make a period, you need to hit enter. Every time you want to make a new paragraph, you press enter twice. These sentences must be on separate lines, but without a blank line between them.